notebooks/samples/104 - Train, Test, Evaluate for Regression with Auto Imports Dataset.ipynb

```diff
@@ -0,0 +1,310 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 104 - Train, Test, Evaluate for Regression with Auto Imports Dataset\n",
+    "\n",
+    "This sample notebook is based on the Gallery [Sample 6: Train, Test, Evaluate for Regression: Auto Imports Dataset](https://gallery.cortanaintelligence.com/Experiment/670fbfc40c4f44438bfe72e47432ae7a) for AzureML Studio. This experiment demonstrates how to build a regression model to predict the automobile's price. The process includes training, testing, and evaluating the model on the Autom Imports data set.\n",
+    "\n",
+    "This sample demonstrates the use of several members of the mmlspark library:\n",
+    "- `TrainRegressor`: [TrainRegressor](http://mmlspark.azureedge.net/docs/pyspark/TrainRegressor.html)\n",
+    "- `SummarizeData`: [SummarizeData](http://mmlspark.azureedge.net/docs/pyspark/SummarizeData.html)\n",
+    "- `CleanMissingData`: [CleanMissingData](http://mmlspark.azureedge.net/docs/pyspark/CleanMissingData.html)\n",
+    "- `ComputeStatistics`: [ComputeStatistics](http://mmlspark.azureedge.net/docs/pyspark/ComputeStatistics.html)\n",
+    "- `FindBestModel`: [FindBestModel](http://mmlspark.azureedge.net/docs/pyspark/FindBestModel.html)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, import the pandas package"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Declare the schema for the data that will be read in. Allow all fields to be nullable, so that missing values can be handled appropriately, such as replacing them with the mean or median value for that column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from pyspark.sql.types import LongType, StringType, DoubleType, StructType, StructField\n",
+    "\n",
+    "colSchema = ((\"symboling\", LongType), (\"normalized-losses\", DoubleType), (\"make\", StringType), (\"fuel-type\", StringType),\n",
```

Comment:
Does the schema need to be declared explicity? Wouldn't pandas csv reader be able to infer it automatically?

Reply:
Yes, the pandas csv reader infers the schema. However, the missing values cause an exception when converting from pandas to spark. Even though pandas detected that the columns were nullable numberic types (verified by printing the schema), spark couldn't handle the missing values. The only solution I could find was to explicitly declare the numeric columns as the appropriate datatype and set nullable to True.